SU2_CFD/src/solvers/CIncNSSolver.cpp

```diff
@@ -97,6 +98,43 @@ void CIncNSSolver::Preprocessing(CGeometry *geometry, CSolver **solver_container
 
   ComputeVorticityAndStrainMag<1>(*config, iMesh);
 
+  /*--- Compute recovered pressure and temperature for streamwise periodic flow ---*/
+  if (config->GetKind_Streamwise_Periodic() != NONE) {
+
+    /*--- Define and initialize helping variables ---*/
+    su2double dot_product, Pressure_Recovered, Temperature_Recovered;
+
+    const su2double delta_p  = config->GetStreamwise_Periodic_PressureDrop() / config->GetPressure_Ref();
+
+    /*--- Reference node on inlet periodic marker to compute relative distance along periodic translation vector. ---*/
+    const su2double* ReferenceNode = geometry->GetStreamwise_Periodic_RefNode();
+
+    /*--- Compute square of the distance between the 2 periodic surfaces. ---*/
+    const su2double norm2_translation = GeometryToolbox::SquaredNorm(nDim, config->GetPeriodic_Translation(0));
+
+    /*--- Compute recoverd pressure and temperature for all points ---*/
+    for (auto iPoint = 0ul; iPoint < nPoint; iPoint++) {
+
+      /*--- First, compute helping terms based on relative distance (0,l) between periodic markers ---*/
+      dot_product = 0.0;
+      for (unsigned short iDim = 0; iDim < nDim; iDim++)
+        dot_product += fabs( (geometry->nodes->GetCoord(iPoint,iDim) - ReferenceNode[iDim]) * config->GetPeriodic_Translation(0)[iDim]);
+
+      /*--- Second, substract/add correction from reduced pressure/temperature to get recoverd pressure/temperature ---*/
+      Pressure_Recovered = nodes->GetSolution(iPoint, 0) - delta_p / norm2_translation * dot_product;
+      nodes->SetStreamwise_Periodic_RecoveredPressure(iPoint, Pressure_Recovered);
+
+      /*--- InnerIter > 0 as otherwise MassFlow in the denominator would be zero ---*/
+      if (energy && InnerIter > 0) {
+        Temperature_Recovered  = nodes->GetSolution(iPoint, nDim+1);
```

Comment:
Yes that is good point. Always good to avoid these implicit accesses. There are some other places where I did that prob -> so I'll sweep through and change those as well if possible